crates/stackable-operator/src/commons/telemetry.rs

```diff
@@ -0,0 +1,87 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::time::Duration;
+
+// NOTE (@Techassi): We might want to rename this to OpenTelemetryConfig to
+// avoid confusion about business telemetry.
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TelemetryConfig {
+    pub metrics: MetricsConfig,
+    pub traces: TracesConfig,
+    pub logs: LogsConfig,
+
+    /// Global default exporter configuration.
+    pub defaults: ExporterConfig,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetricsConfig {
+    /// Enables the export of metrics.
+    #[serde(default = "r#true")]
+    pub enabled: bool,
+
+    /// Overides the global exporter config.
+    pub exporter: Option<ExporterConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TracesConfig {
+    /// Enables the export of traces.
+    #[serde(default = "r#true")]
+    pub enabled: bool,
+
+    /// Overides the global exporter config.
+    pub exporter: Option<ExporterConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LogsConfig {
+    /// Enables the export of logs.
+    #[serde(default = "r#true")]
+    pub enabled: bool,
+
+    /// Overides the global exporter config.
+    pub exporter: Option<ExporterConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExporterConfig {
```

Comment:
Is this list of Configs (Telemetry, Metrics, Traces, Logs, Exporter) exhaustive? (I can't think of any other relevant element). If so, is it worth "closing" it somehow, e.g. by associating it with an enum?

Reply:
Sure, we can exclude the structs from this PR. I can move the file to a separate branch to keep it around for when we re-visit that topic.

Reply:
Removed in 67062be, changes saved in https://github.com/stackabletech/operator-rs/tree/feat/otel-crd-structs